simln-lib/src/lib.rs

```diff
@@ -1559,4 +1593,39 @@ mod tests {
             payment_interval
         );
     }
+
+    #[tokio::test]
+    async fn test_validate_zero_amount_no_valid() {
+        let mut node_1 = test_utils::create_nodes(1, 100_000)
+            .first()
+            .unwrap()
+            .0
+            .clone();
+        node_1.features.set_keysend_optional();
+        let mut node_2 = test_utils::create_nodes(1, 0).first().unwrap().0.clone();
+        node_2.features.set_keysend_optional();
```

Comment:
Rather than setting keysend here, let's update `create_nodes` to set this feature bit? We just didn't need it in the past, which is why it's an empty vector.

Reply:
ok, maybe it would be better to open another PR doing this? because several places call `create_nodes` that are not related with this PR. WDYT?

Reply:
I don't think it'll break any existing tests, so I think we can go ahead in this PR? I was thinking just set it inside of `create_nodes` so that fn signature doesn't change so it would be a small change.

_But_ if I'm wrong (always the most likely outcome 😅 ) and it breaks a bunch of stuff then yes let's do a followup!